File: little-bee/controller/router/fsu_type.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_FSU_TYPE = "/config/fsu-type"
const URL_ROUTER_FSU_TYPE_BY_ID = "/config/fsu-type/:fsu-type-id"
const URL_ROUTER_FSU_TYPE_QUERY = "/config/fsu-type/query"
const URL_ROUTER_FSU_TYPE_ALL = "/config/fsu-type/all"

func FsuTypeRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_FSU_TYPE, config.CreateFsuType)
	r.GET(URL_ROUTER_FSU_TYPE_BY_ID, config.RetrieveFsuType)
	r.PUT(URL_ROUTER_FSU_TYPE, config.UpdateFsuType)
	r.DELETE(URL_ROUTER_FSU_TYPE_BY_ID, config.DeleteFsuType)
	r.PUT(URL_ROUTER_FSU_TYPE_BY_ID, config.DeleteFsuTypeWithFlag)
	r.POST(URL_ROUTER_FSU_TYPE_QUERY, config.QueryFsuType)
	r.GET(URL_ROUTER_FSU_TYPE_ALL, config.GetAllFsuTypes)
	return r
}

File: frontend/console/src/api/fsu-type.js
import Axios from "@/http";

export function createFsuType(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/fsu-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveFsuType(fsuTypeId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/fsu-type/${ fsuTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateFsuType(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/fsu-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteFsuType(fsuTypeId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/fsu-type/${ fsuTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteFsuTypeWithFlag(fsuTypeId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/fsu-type/${ fsuTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllFsuTypes() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/fsu-type/all').then(res => {
            resolve(res)
        })
    })
}

export function queryFsuType (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/fsu-type/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}